### ISSUE TITLE
Docker downgrade failure

### ISSUE THREAD

**Issue by jpic** [openshift/openshift-ansible#1713]
The first run of the playbook was fine, but this is how a rerun fails:

```
TASK: [docker | Downgrade docker if necessary] ********************************
failed: [master-0.yourlabs.org] => {"changed": true, "cmd": ["yum", "downgrade", "-y", "docker-1.8.2"], "delta": "0:00:04.171900", "end": "2016-04-06 11:14:36.374244", "rc": 1, "start": "2016-04-06 11:14:32.202344", "warnings": ["Consider using yum module rather than running yum"]}
stderr: Error: Package: origin-node-1.1.6-0.git.0.ef1caba.el7.centos.x86_64 (@maxamillion-origin-next)
           Requires: docker-io >= 1.9.1
           Removing: docker-1.9.1-25.1.origin.el7.centos.x86_64 (@maxamillion-origin-next)
               docker-io = 1.9.1-25.1.origin.el7.centos
           Downgraded By: docker-1.8.2-10.el7.centos.x86_64 (extras)
               docker-io = 1.8.2-10.el7.centos
           Available: docker-1.8.2-7.el7.centos.x86_64 (extras)
               docker-io = 1.8.2-7.el7.centos
           Available: docker-1.8.2-8.el7.centos.x86_64 (extras)
               docker-io = 1.8.2-8.el7.centos
           Available: docker-1.9.1-25.el7.centos.x86_64 (extras)
               docker-io = 1.9.1-25.el7.centos
stdout: Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: repo.bigstepcloud.com
 * extras: centos.hyve.com
 * updates: centos.hyve.com
Resolving Dependencies
--> Running transaction check
---> Package docker.x86_64 0:1.8.2-10.el7.centos will be a downgrade
---> Package docker.x86_64 0:1.9.1-25.1.origin.el7.centos will be erased
--> Finished Dependency Resolution
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

FATAL: all hosts have already failed -- aborting
```

This is a fresh install started an hour ago:

origin v1.1.6
kubernetes v1.2.0-36-g4a3f9c5
etcd 2.2.5

openshift-ansible version: f13efb21bb48a2202a09d70bf0a0103461b22950

centos version: 7.2

**Comment by detiber** [comment]
@sdodson do we need to change the downgrade to only apply to origin < 1.1.6 ?

**Comment by sdodson** [comment]
Should be fixed in master now by only downgrading when origin < 1.1.4, can you verify?

**Comment by zhwenh** [comment]
open  roles\docker\tasks\main.yml
look this ： 
  - name: Downgrade docker if necessary
  command: "{{ ansible_pkg_mgr }} downgrade -y docker-{{ docker_version }}"

“downgrade” modified to “install"  
THATS OK!

**Comment by jpic** [comment]
Works for me, keep up the great work !

**Comment by lvthillo** [comment]
Same issue. I have a cluster 1.1.3 and I want to add a node to the existing cluster. I want to rerun the playbook but I don't see what to change. my main.yml looks like this:

```
# tasks file for docker
- name: Install docker
  action: "{{ ansible_pkg_mgr }} name=docker state=present"
  when: not openshift.common.is_atomic | bool

- name: enable and start the docker service
  service:
    name: docker
    enabled: yes
    state: started
  register: start_result

- set_fact:
    docker_service_status_changed: start_result | changed

- include: udev_workaround.yml
  when: docker_udev_workaround | default(False)
```

**Comment by lvthillo** [comment]
Okay, updated the repo and set the env_var for docker version. Works now.

**Comment by lvthillo** [comment]
@sdodson How can I set origin < 1.1.4. It's refused when I put this in my playbook.

**Comment by sdodson** [comment]
@lvthillo Can you try the latest code from master? What you've pasted has been significantly updated over the past week or two.

**Comment by lvthillo** [comment]
I cloned the new repository.
`git clone https://github.com/openshift/openshift-ansible`

In my /etc/ansible/hosts
```
product_type=openshift
deployment_type=origin
docker_version=1.8.2
origin_version=1.1.3
```
TASK install and enable docker:
version is 1.8.2

```
TASK: [openshift_node | Install Node package] *********************************
ok: [ip-172-xx-xx-xx]
ok: [ip-172-xx-xx-xx]
ok: [ip-172-xx-xx-xx]
changed: [ip-172-xx-xx-xx]
```
After the task above my docker version is changing to 1.9.1 on then ew node

**Comment by lvthillo** [comment]
@sdodson I performed the same step as @jpic . But after rerunning there isn't a downgrade if I change downgrade by install. (it gives the same error when I don't change it).

**Comment by sdodson** [comment]
The only version of Origin that's installed by the installer right now requires docker-1.9.1. So it's expected that you're running docker-1.9.1

**Comment by lvthillo** [comment]
@sdodson it's not possible to downgrade that package so I can still use docker 1.8.2?

**Comment by sdodson** [comment]
@lvthillo The problem is that COPR only keeps the latest build, so unless you've mirrored older builds of origin then origin 1.1.6 is the only version that's available and that requires docker-1.9.1.
